### PR TITLE
feat(driver_test.go): add test around docker driver config

### PR DESCRIPTION
**Pull request by vdice**
# What does this change
* adds a test around Docker driver config based on runtime extension data

Depends on https://github.com/cnabio/cnab-go/pull/206 or some variant thereof

TODO:
  - [ ] remove branch override in go.mod

# What issue does it fix
Addresses a `TODO` lingering in code

# Notes for the reviewer
Am I go mod-ing right?!  All of the updated deps are coming from my branch override, right?  Or is something up with my local Go cache/env that's bringing all of these in?

# Checklist
- [x] Unit Tests
- [x] Documentation
- [x] Schema (porter.yaml)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -39,10 +38,10 @@ require (
 	github.com/pivotal/image-relocation v0.0.0-20191111101224-e94aff6df06c
 	github.com/pkg/errors v0.8.1
 	github.com/spf13/afero v1.2.2
-	github.com/spf13/cobra v0.0.5
+	github.com/spf13/cobra v0.0.6
 	github.com/spf13/pflag v1.0.5
-	github.com/spf13/viper v1.3.2
-	github.com/stretchr/testify v1.4.0
+	github.com/spf13/viper v1.4.0
+	github.com/stretchr/testify v1.5.1
```
Comment (carolynvs) [inline go.mod:44]: cnab-go pulls in a higher version of testify and packr. The packr bump pulled in a new dependency (envy), etc... It was a cascade from bumping cnab-go and its dependencies.

**Comment by carolynvs** [comment]
I don't see what caused you to need the changes in the PR to cnab-go? Can you help explain what in Porter made you need the change? I'm sure there was something that made this originally hard to work with.

**Comment by vdice** [comment]
> I don't see what caused you to need the changes in the PR to cnab-go?

I added some notes in https://github.com/cnabio/cnab-go/pull/206#issue-407561707 but found we'd hit panics when applying configuration options to an otherwise uninitialized Driver instance (meaning, not actually running `exec` on the driver, where much of the initialization occurs), because the `containerCfg` and `containerHostCfg` members were also uninitialized (invalid references).  I was hoping to support the ability to do the aforementioned, and so thought to change those fields to actual values.

File: pkg/cnab/provider/driver_test.go [hunk 2]
```diff
@@ -52,6 +53,28 @@ func TestNewDriver_Docker(t *testing.T) {
...
+
+		dockerish, ok := driver.(*docker.Driver)
+		assert.True(t, ok)
+
+		err = dockerish.ApplyConfigurationOptions()
+		assert.NoError(t, err)
+
+		containerHostCfg, err := dockerish.GetContainerHostConfig()
+		require.NoError(t, err)
+		require.Equal(t, true, containerHostCfg.Privileged)
```
Comment (carolynvs) [inline pkg/cnab/provider/driver_test.go:78]: This is perfect. Thanks for your work getting the plumbing into cnab-go to support it.

Can you add a converse test to verify that when we don't have privileged that it's not set on the driver? Just to check for us hard coding it accidentally like it was originally.

**Pull request merged**